assets/js/blocks/reviews-by-product/block.js

```diff
@@ -1,283 +1,90 @@
 /**
  * External dependencies
  */
-import { escapeHTML } from '@wordpress/escape-html';
-import { __, _n, sprintf } from '@wordpress/i18n';
-import {
-	BlockControls,
-	InspectorControls,
-} from '@wordpress/editor';
-import {
-	Button,
-	PanelBody,
-	Placeholder,
-	SelectControl,
-	TextControl,
-	ToggleControl,
-	Toolbar,
-	withSpokenMessages,
-	SearchListItem,
-} from '@wordpress/components';
-import { Component, Fragment } from '@wordpress/element';
-import { compose } from '@wordpress/compose';
-import PropTypes from 'prop-types';
 import apiFetch from '@wordpress/api-fetch';
+import classNames from 'classnames';
+import { Component } from '@wordpress/element';
 import { debounce } from 'lodash';
+import PropTypes from 'prop-types';
+import { stringifyQuery } from '@woocommerce/navigation';
 
 /**
  * Internal dependencies
  */
-import ProductControl from '../../components/product-control';
-import ReviewsByProductIcon from '../../components/icons/reviews-by-product';
-import { renderReviewsByProductPlaceholder } from './utils';
+import { renderReview } from './utils';
 
 /**
  * Component to handle edit mode of "Reviews by Product".
  */
 class ReviewsByProduct extends Component {
-
 	constructor() {
 		super( ...arguments );
 		this.state = {
-			product: false,
-			loaded: false,
+			reviews: [],
 		};
 
-		this.debouncedGetProduct = debounce( this.getProduct.bind( this ), 200 );
+		this.debouncedGetReviews = debounce( this.getReviews.bind( this ), 200 );
 	}
 
 	componentDidMount() {
-		this.getProduct();
+		this.getReviews();
 	}
 
 	componentDidUpdate( prevProps ) {
-		if ( prevProps.attributes.productId !== this.props.attributes.productId ) {
-			this.debouncedGetProduct();
+		if (
+			prevProps.attributes.orderby !== this.props.attributes.orderby ||
+			prevProps.attributes.perPage !== this.props.attributes.perPage ||
+			prevProps.attributes.productId !== this.props.attributes.productId
+		) {
+			this.getReviews();
 		}
 	}
 
-	getProduct() {
-		const { productId } = this.props.attributes;
+	getReviews() {
+		const { attributes } = this.props;
+		const { orderby, perPage, productId } = attributes;
+
 		if ( ! productId ) {
 			// We've removed the selected product, or no product is selected yet.
-			this.setState( { product: false, loaded: true } );
 			return;
 		}
+
+		const payload = stringifyQuery( { order_by: orderby, per_page: perPage, product: productId } );
+
 		apiFetch( {
-			path: `/wc/blocks/products/${ productId }`,
+			path: `/wc/v3/products/reviews${ payload }`,
```

Comment:
This can be done in a follow up PR and needs a note, but this API is not open to authors so we'll need to extend it with a blocks endpoint for this to work.

Reply:
Thank you, added a note in #658.